wagtailio/areweheadlessyet/models.py

```diff
@@ -0,0 +1,42 @@
+from django.db import models
+from wagtail.admin.edit_handlers import FieldPanel, MultiFieldPanel, StreamFieldPanel
+from wagtail.core.fields import RichTextField, StreamField
+from wagtail.core.models import Page
+
+from wagtailio.areweheadlessyet.blocks import HomePageBlock
+from wagtailio.utils.models import CrossPageMixin, SocialMediaMixin
+
+
+class AreWeHeadlessYetHomePage(Page, SocialMediaMixin, CrossPageMixin):
+    THUMBS_UP = "thumbs up"
+    THUMBS_DOWN = "thumbs down"
+    ICON_CHOICES = [
+        (THUMBS_UP, THUMBS_UP),
+        (THUMBS_DOWN, THUMBS_DOWN),
+    ]
+    strapline_icon = models.CharField(
+        max_length=15,
+        choices=ICON_CHOICES,
+        default=THUMBS_UP,
+    )
+    strapline_text = RichTextField(features=["bold", "italic"])
+    body = StreamField(HomePageBlock())
+
+    content_panels = Page.content_panels + [
+        MultiFieldPanel(
+            [
+                FieldPanel("strapline_icon"),
+                FieldPanel("strapline_text"),
+            ],
+            "strapline",
+        ),
+        StreamFieldPanel("body"),
+    ]
+
+    promote_panels = (
+        Page.promote_panels + SocialMediaMixin.panels + CrossPageMixin.panels
+    )
+
+    max_count = 1
+    parent_page_types = ["wagtailcore.Page"]
```

Comment:
question: what is the purpose for this? could use a comment for future devs

Reply:
The `parent_page_types` line is added so that we can only create the AreWeHeadlessYetHomepage at the root level (another site instance).
The `max_count` line is here for ensuring that we are can add one and only one AreWeHeadlessYetHomePage.

Does it make sense?

Reply:
Hey @Tijani-Dia 👍🏼 all makes sense to me. As I mentioned, worth adding a comment to that effect

Reply:
Added!